packages/knip/src/plugins/eleventy/index.ts

```diff
@@ -18,6 +21,14 @@ const PRODUCTION_ENTRY_FILE_PATTERNS = ['posts/**/*.11tydata.js', '_data/**/*.{j
 const findEleventyDependencies: GenericPluginCallback = async (configFilePath, options) => {
   const { config } = options;
 
+  let localConfig = await load(configFilePath);
+  if (typeof localConfig === 'function') localConfig = await localConfig(new DummyEleventyConfig());
+
+  const result = { ...localConfig, ...defaultEleventyConfig };
+
+  // Data directory pattern:
+  toEntryPattern(`${join(result.dir.input, result.dir.data)}/**/*.js`);
```

Comment:
Here is the new pattern that we want to be considering. How should I factor this into the existing return logic below?

Reply:
I've written some comments in the plugin template:

https://github.com/webpro/knip/blob/d8babee3fbb54d2e3fdf2fdb8f7dd55f52970d4d/packages/knip/src/plugins/_template/index.ts

The Playwright plugin seems similar to me:
 https://github.com/webpro/knip/blob/d8babee3fbb54d2e3fdf2fdb8f7dd55f52970d4d/packages/knip/src/plugins/playwright/index.ts

Two differences:

- The playwright plugin has an extra exported function to be reused by the playwright-ct plugin (you don't necessarily need to split it like that).
- The 11ty plugin should use `toProductionEntryPattern` (instead of `toEntryPattern`, sorry forgot about that before), as there's a difference between "dev dependencies" (like test frameworks target e.g. `*.test.ts` entry files) and "production dependencies" (plugins for Next.js, Remix, Gatsby use that too).

Eager to know whether the template comments together with https://knip.dev/guides/writing-a-plugin makes sense, and where it should be improved :)

Reply:
Thank you. How does it look now?